### PR TITLE
Enable drag and drop for step listview

### PR DESCRIPTION
**Pull request by Kayvv**

### CODE REVIEW

File: mapclientplugins/geometryfitter/view/geometryfitterwidget.py [hunk 1]
```diff
@@ -410,6 +423,8 @@ def _updateDisplayWidgets(self):
         self._ui.displaySurfacesExterior_checkBox.setChecked(self._model.isDisplaySurfacesExterior())
         self._ui.displaySurfacesTranslucent_checkBox.setChecked(self._model.isDisplaySurfacesTranslucent())
         self._ui.displaySurfacesWireframe_checkBox.setChecked(self._model.isDisplaySurfacesWireframe())
+        self._ui.displayRMSError_lineEdit.setText("TEST!")
+        self._ui.displayMaxError_lineEdit.setText("TEST22!")
```
Comment (hsorby) [inline mapclientplugins/geometryfitter/view/geometryfitterwidget.py:427]: These "values" will come from the model in the future `self._model.rmsError()` and `self._model.maxError()`. Might as well have these in place now.

File: mapclientplugins/geometryfitter/view/geometryfitterwidget.py [hunk 2]
```diff
@@ -245,34 +248,44 @@ def _buildStepsList(self):
...
+        self._ui.steps_listWidget.show()
         self._updateFitterStepWidgets()
 
+    def _onStepsListItemChanged(self, prevRow, newRow):
+        """
+        For steps list drag and drop event.
+        Update the order of steps in fitterSteps.
+        """ 
+        if newRow != prevRow:
+            fitterSteps = self._fitter.getFitterSteps()
```
Comment (hsorby) [inline mapclientplugins/geometryfitter/view/geometryfitterwidget.py:271]: Why bother getting fitter steps if you are not going to use them.

**Review by hsorby — CHANGES_REQUESTED** [review]

**Comment by hsorby** [comment]
@rchristie has implemented the fitting error calculation in scaffoldfitter now (and it has been merged in), that means this PR can be finished off @Kayvv.

**Comment by Kayvv** [comment]
> @rchristie has implemented the fitting error calculation in scaffoldfitter now (and it has been merged in), that means this PR can be finished off @Kayvv.

I will start a new PR for display_error. This is an old PR for drag_drop, sorry I merged display_error here just for update and testing.

**Comment by hsorby** [comment]
> I will start a new PR for display_error. This is an old PR for drag_drop, sorry I merged display_error here just for update and testing.

Are you going to remove the display_error parts that don't belong in this PR?

**Comment by Kayvv** [comment]
Yep, I’ll merge the main branch in again once display_error has been merged into main and fix the conflicts. I found another bug here today (the display group doesn’t work after reordering the step list) so probably needs more testing and debugging. I this this PR has less priority than display_error.

**Review by hsorby — APPROVED** [review]
Looks great and works well, thanks.

**Pull request merged**